zap/src/test/java/org/zaproxy/zap/utils/ZapXmlConfigurationUnitTest.java

```diff
@@ -43,6 +48,88 @@ class ZapXmlConfigurationUnitTest {
                     + "    </d>\n"
                     + "</a>\n";
 
+    @Test
+    void shouldBeInstantiatedWithDelimterParsingDisabled() {
+        // Given
+        ZapXmlConfiguration config = new ZapXmlConfiguration();
+        // When
+        boolean isParsingDisabled = config.isDelimiterParsingDisabled();
+        // Then
+        assertThat(isParsingDisabled, is(equalTo(true)));
+    }
+
+    @Test
+    void shouldSetVariousPropertiesAsExpected()
+            throws ConfigurationException, UnsupportedEncodingException {
+        // Given
+        ZapXmlConfiguration config = new ZapXmlConfiguration();
+        String aString = "Some string 🙂.";
+        String bString = "";
+        String cString = null;
+        int aInt = 120;
+        boolean aBoolean = true;
+        // When
+        config.setProperty("aString", aString);
+        config.setProperty("bString", bString);
+        config.setProperty("cString", cString);
+        config.setProperty("aInt", aInt);
+        config.setProperty("aBoolean", aBoolean);
+        List<Object> aStringList = config.getList("aString");
+        List<Object> bStringList = config.getList("bString");
+        List<Object> cStringList = config.getList("cString");
+        List<Object> aIntList = config.getList("aInt");
+        int aActualInt = config.getInt("aInt");
+        List<Object> aBooleanList = config.getList("aBoolean");
+        boolean aActualBoolean = config.getBoolean("aBoolean");
```

Comment:
IMO the gets should be in the `Then` part, that's for validation not the action that's being verified.